### DISCUSSION TITLE
How to automatically back up important data?

### DISCUSSION THREAD

**Discussion by cukewu** [dani-garcia/vaultwarden#5050] [Q&A] [answered]
I deployed Vaultwarden on Synology according to the method described in the wiki documentation and enabled OTP two-factor authentication. It is currently running stably in Docker, and I have mapped the persistent data location to a local hard drive on the NAS. I would like to ask for advice on how to automatically back up, or which files I should back up to ensure that Vaultwarden's important database is synchronized to another location. This is to prevent data loss in case of a hard drive failure on the NAS. (From what I’ve checked, the current documentation mentions manual export via JSON or CSV from the web interface.)

**Comment by cukewu** [comment]
thks. I  found it
https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault

**Reply by cukewu** [reply]
The operation is more complicated. Can I directly copy the /data directory?

**Reply by Gerardv514** [reply, marked as answer]
Can you just backup the docker\vaultwarden folder? I believe that is where all of the data is, including the database. 

Additionally, you’ll need to go into container manager, and export the container otherwise you would have to rebuild the container and map it to the docker data folder.

**Reply by WirlyWirly** [reply]
Thanks, I'm using a Vaultwarden container for my family, but I was scratching my head a bit thinking about how I'm going to do backups. I thought there'd be more to it than just copying the folder, such as permission issues.

The structure is a little more complicated than the single .kpdx that my own KeePass database uses.

**Reply by Gerardv514** [reply]
Additionally, I don't know if you have capability to run the snapshot application, but that has recently saved me when there was a database corruption. Recently Bitwarden had released their new native apps, and I had upgraded my iOS device. In my particular situation, I use an organization, and Vaultwarden and the new native app had caused an issue; therefore corrupting my whole vault. I was able to restore in the matter of 1.5 minutes the snapshot of the docker\vaultwarden folder and I was back up and running. From there I did not use the new native app until Vaultwarden corrected their code. In addition to the snapshots, I have HyperBackup where I do backups to an external drive and an offsite location. So far no issues even with the restore.

I have even taken the live/production docker\vaultwarden folder and created a copy, adding the word test to it. I created an additional container, separate sub domain name and I am able to run a test instance where I would mess around with.
